rust/xaynet-analytics/src/database/controller_data/adapter.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::{anyhow, Error, Result};
+use isar_core::object::{
+    data_type::DataType,
+    isar_object::{IsarObject, Property},
+    object_builder::ObjectBuilder,
+};
+use std::vec::IntoIter;
+
+use crate::database::common::{FieldProperty, IsarAdapter, SchemaGenerator};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ControllerDataAdapter {
+    pub time_data_sent: String,
+}
+
+impl ControllerDataAdapter {
+    pub fn new(time_data_sent: String) -> Self {
+        Self { time_data_sent }
+    }
+}
+
+impl<'ctrl> IsarAdapter<'ctrl> for ControllerDataAdapter {
+    fn into_field_properties() -> IntoIter<FieldProperty> {
+        vec![FieldProperty::new(
+            "time_data_sent".to_string(),
```

Comment:
you're right. i think a possible very nice solution is to have a macro that automates the whole implementation of a `IsarAdapter` for each data model, since it's always the same and it would avoid some possible future bugs (cause you indeed need the same name in both methods). But unfortunately i won't have time to work on this. I'll add it to the handover stuff, I think it shouldn't be too difficult for one of you peeps